src/utils.rs

```diff
@@ -55,6 +56,35 @@ pub fn encode_with_code(output: &mut String, text: &str) {
     }
 }
 
+static UNICODE_CHARS_MAP: phf::Map<char, &str> = phf_map! {
+    '“' => "\"",
+    '”' => "\"",
+    '‘' => "\'",
+    '’' => "\'",
+    '—' => "--"
+};
+
+/// Normalize the mistakenly inputted Unicode character
+/// to the corresponding ASCII character.
+///
+/// For the table what characters this function will convert,
+/// you can refer to [`UNICODE_CHARS_MAP`].
+///
+/// Time complexity of this is `O(n)`.
+pub fn normalize_unicode_chars(inputs: &str) -> String {
+    let mut output = String::with_capacity(inputs.len());
```

Comment:
This feels a bit expensive, and in majority of the cases it's not needed, so maybe we can make it return `Cow<str>` and add a fast path like
```rust
if input.is_ascii() {
  return Cow::Borrowed(input);
}
```

WDYT?

It's probably not a big deal, though, since very few people really put a long code for eval bot.